.github/workflows/go.yml

```diff
@@ -2,24 +2,29 @@ name: Go
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 jobs:
-
-  build:
+  lint-and-test:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+      - name: lint
+        uses: golangci/golangci-lint-action@v3.1.0
```

Comment:
Do we need to do something extra here to lint recursively or is that the default behavior?

Reply:
I honestly can't tell from documentation. I could push a commit that breaks something in example and find out.